src/selectors/getThoughtById.ts

```diff
@@ -2,7 +2,10 @@ import State from '../@types/State'
 import ThoughtId from '../@types/ThoughtId'
 
 /**
- * Gets a Thought by its ThoughtId.
+ * Gets a Thought by its ThoughtId. Can be undefined if the Thought is not in the state.
+ * The only time state.thoughts.thoughtIndex[id] is only undefined in rare
+ * circumstances after an async operation or unmount, e.g. CSSTransition,
+ * but it is possible nonetheless and should be typed accordingly.
```

Comment:
since, I've left a couple `see getThoughtById` comments where I've found things not to be clear, I felt that adding why it could be undefined, makes sense
if it does make sense, maybe I should even put the issue link here too, right?

Reply:
Good idea